dspace-server-webapp/src/test/java/org/dspace/app/rest/DiscoveryRestControllerIT.java

```diff
@@ -3582,4 +3582,139 @@ public void discoverSearchObjectsWorkflowConfigurationTest() throws Exception {
 
     }
 
+    @Test
+    public void discoverSearchObjectsTestForWithdrawnOrPrivateItemsNonAdmin() throws Exception {
+        //We turn off the authorization system in order to create the structure as defined below
+        context.turnOffAuthorisationSystem();
+
+        //** GIVEN **
+        //1. A community-collection structure with one parent community with sub-community and two collections.
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                                          .withName("Parent Community")
+                                          .build();
+        Community child1 = CommunityBuilder.createSubCommunity(context, parentCommunity)
+                                           .withName("Sub Community")
+                                           .build();
+        Collection col1 = CollectionBuilder.createCollection(context, child1).withName("Collection 1").build();
+        Collection col2 = CollectionBuilder.createCollection(context, child1).withName("Collection 2").build();
+        //2. Two public items that are readable by Anonymous with different subjects and one private item
+        Item publicItem1 = ItemBuilder.createItem(context, col1)
+                                      .withTitle("Test")
+                                      .withIssueDate("2010-10-17")
+                                      .withAuthor("Smith, Donald")
+                                      .withSubject("ExtraEntry")
+                                      .build();
+
+        Item publicItem2 = ItemBuilder.createItem(context, col2)
+                                      .withTitle("WithdrawnTest 2")
+                                      .withIssueDate("1990-02-13")
+                                      .withAuthor("Smith, Maria").withAuthor("Doe, Jane")
+                                      .withSubject("ExtraEntry")
+                                      .withdrawn()
+                                      .build();
+
+        Item publicItem3 = ItemBuilder.createItem(context, col2)
+                                      .withTitle("Private Test item 2")
+                                      .withIssueDate("2010-02-13")
+                                      .withAuthor("Smith, Maria").withAuthor("Doe, Jane")
+                                      .withSubject("AnotherTest").withSubject("ExtraEntry")
+                                      .makeUnDiscoverable()
+                                      .build();
+
+
+        String query = "Test";
+        //** WHEN **
+        //An anonymous user browses this endpoint to find the the withdrawn or private objects in the system
+        //With a query stating 'Test'
+        getClient().perform(get("/api/discover/search/objects")
+            .param("configuration", "showUnDiscoverableItems")
```

Comment:
The name of this configuration was initially confusing to me.  I *thought* that `showUnDiscoverableItems` would return *both* public and undiscoverable items (as the word "show" to me implied it would return both).  However, after looking at the tests here, I realize I was wrong in that assumption...this *only* returns the "undiscoverable" items (excluding the public ones).

That behavior seems OK to me. But, could we rename this configuration slightly to make it clearer?  Maybe call it simply `undiscoverable`, or maybe `restricted`?   That would also align it better with other configurations like `workflow` and `workspace`, which just return Items with that status.

Reply:
Good suggestion.  I was initially thinking in terms of both public and undiscoverable items so the name made sense.  Will change.

I notice that separate endpoints for discovering withdrawn and private items are listed on the estimation spreadsheet, and that there's a question about whether these should be combined into a single endpoint, as is done here. Obviously, if separate configurations are needed that's an easy change. It could be added to this PR if needed.